### PR TITLE
Implement zlib-stream compression

### PR DESCRIPTION
**Pull request by ArsenArsen**
### Prerequisites
* [x] I have read and understood the [contribution guidelines](CONTRIBUTING.md)
* [x] This pull request follows the code style of the project
* [x] I have tested this feature thoroughly
  * [Proof of testing](https://nothing.domains/978a7aa.png)

**Issues Fixed:** Non applicable

### Changes Proposed in this Pull Request
* Make zlib-stream compression work
* Use IOUtils instead of a Stream<String> for reading the decompressed data

### CODE REVIEW

**Comment by GrandPanda** [comment]
The query parameter name is incorrect which probably makes all previous testing irrelevant.

**Comment by ArsenArsen** [comment]
Line 233 triggers and the onWebsocketText method called next has the correct payload.

**Comment by GrandPanda** [comment]
Binary frames can still be received because we have `compress:true` in the identify packet. Actually enable zlib stream by using the proper parameter key (`compress`) and enjoy your exceptions.

File: src/main/java/sx/blah/discord/api/internal/DiscordWS.java [hunk 1]
```diff
@@ -37,7 +38,8 @@
 import sx.blah.discord.util.LogMarkers;
 
 import java.io.BufferedReader;
-import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
```
Comment (danthonywalker) [inline src/main/java/sx/blah/discord/api/internal/DiscordWS.java]: Useless

File: src/main/java/sx/blah/discord/api/internal/DiscordWS.java [hunk 2]
```diff
@@ -227,10 +235,11 @@ public void onWebSocketError(Throwable cause) {
 
 	@Override
 	public void onWebSocketBinary(byte[] payload, int offset, int len) {
-		BufferedReader reader = new BufferedReader(new InputStreamReader(new InflaterInputStream(new ByteArrayInputStream(payload, offset, len))));
-		onWebSocketText(reader.lines().collect(Collectors.joining()));
-		try {
-			reader.close();
+		ByteArrayOutputStream out = new ByteArrayOutputStream(payload.length * 2);
```
Comment (danthonywalker) [inline src/main/java/sx/blah/discord/api/internal/DiscordWS.java]: This should be in its own try-with. If write or onWebSocketText throws an exception this won't close. Yes, closing a ByteArrayOutputStream has no real effect, but it's best practice in my opinion to always close Closeable resources.

File: src/main/java/sx/blah/discord/api/internal/DiscordWS.java [hunk 3]
```diff
@@ -25,6 +25,7 @@
 import org.eclipse.jetty.websocket.api.WebSocketAdapter;
 import org.eclipse.jetty.websocket.client.ClientUpgradeRequest;
 import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.apache.commons.io.IOUtils;
```
Comment (danthonywalker) [inline src/main/java/sx/blah/discord/api/internal/DiscordWS.java]: Useless

**Comment by GrandPanda** [comment]
This does not implement zlib fragmentation. At this time, v2 does not need to support this compression mode.

**Pull request closed without merging**